presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -414,8 +415,19 @@ public QueryId createQueryId()
             }
             QUERY_STATE_LOG.debug(e, "Query %s failed", session.getQueryId());
 
+            TransactionId transactionId;
+
+            if (session.getTransactionId().isPresent()) {
+                transactionId = session.getTransactionId().get();
+            }
+            else {
+                transactionId = transactionManager.beginTransaction(false);
+                // resolve the catalog session properties in the error path.
+                session = session.beginTransactionId(transactionId, transactionManager, accessControl);
```

Comment:
I'm concerned about this change.  I believe this call can also fail, which would cause an uncaught error in the current code.  Also, with the Dispatcher change coming soon, the dispatcher won't even have connectors registered so, there is no way to "resolve" connector sessions properties during this phase of the execution.

Reply:
So what is the right thing to do here? There are some session properties that are specific to a catalog (they come from the connector but are resolved in catalog scope) and without those e.g. our event listeners do not work. 

Is there another way besides a transaction to resolve those?

Reply:
@dain mentioned that he has a bigger patch that would affect that part of the code. I still would like to revisit this problem after those changes go in because they affect consumers of catalog properties.